### PR TITLE
[ENHANCEMENT] Improve dashboard toolbar UX

### PR DESCRIPTION
**Pull request by AntoineThebaud**
# Description

Improve the dashboard toolbar UX by:
- repositioning the buttons to have less difference between read & edit modes,
- improving its responsiveness: variables take more advantage of the space available + put some limit to the vertical space used when there are many of them, by adding scrolling (basically vertical when not sticky, horizontal when sticky).

_FYI I'm planning to make the JSON view available in the read mode too in a future PR, this is also going to reduce the visual difference between the two modes._ 

# Screenshots

Buttons position:

| Before | After |
:-------------------------:|:-------------------------:
2023-12-13_09h06_33 | 2023-12-15_15h21_20
2023-12-13_09h07_19 | 2023-12-15_15h21_34

Responsiveness:

https://github.com/perses/perses/assets/7058693/bdb22e34-d6cf-44da-9fb4-9469ed9f86de

⚠️ something I didn't take time to improve in this PR is the space used by the edit buttons now that they are on the same row as the breadcrumb: on small screens the latter gets hidden, see demo:

https://github.com/perses/perses/assets/7058693/bd8c01a9-b8d4-4b61-b176-c0d8812f3d29

We should probably group the buttons under a menu in this case as we did for the buttons in the navbar.

# Checklist

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the
  following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `DOC`,`IGNORE`.
- [x] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).

## UI Changes

- [x] Changes that impact the UI include screenshots and/or screencasts of the relevant changes.
- [x] Code follows the [UI guidelines](https://github.com/perses/perses/blob/main/ui/ui-guidelines.md).
- [x] Visual tests are stable and unlikely to be flaky.
  See [Storybook](https://github.com/perses/perses/tree/main/ui/storybook#visual-tests)
  and [e2e](https://github.com/perses/perses/tree/main/ui/e2e#visual-tests) docs for more details. Common issues
  include:
  - Is the data inconsistent? You need to mock API requests.
  - Does the time change? You need to use consistent time values or mock time utilities.
  - Does it have loading states? You need to wait for loading to complete.

### CODE REVIEW

**Review by celian-garcia — APPROVED** [review]
great !

**Comment by AntoineThebaud** [comment]
Working on improving the display on mobile, it's currently less responsive than before..

**Comment by AntoineThebaud** [comment]
Just pushed a small change to keep the timerange controls on the right side in mobile+sticky mode

image

File: ui/dashboards/src/components/DashboardToolbar/DashboardToolbar.tsx [hunk 1]
```diff
@@ -61,30 +61,54 @@ export const DashboardToolbar = (props: DashboardToolbarProps) => {
...
+            display: 'flex',
+            width: '100%',
+            alignItems: 'start',
+            padding: (theme) => theme.spacing(1, 2, 0, 2),
+            flexDirection: isBiggerThanMd ? 'row' : 'column',
+            flexWrap: 'nowrap',
+            gap: 1,
+          }}
+        >
+          <Box flexGrow={1} flexShrink={1}>
```
Comment (Gladorme) [inline ui/dashboards/src/components/DashboardToolbar/DashboardToolbar.tsx:111]:
```suggestion
          <Box flexGrow={1} flexShrink={1} width="100%">
```
Will fix: 
image

Now:
image

**Review by Gladorme — APPROVED** [review]
Cool!

**Pull request merged**